test/tests/functional/pbs_offline_vnodes.py

```diff
@@ -137,21 +179,7 @@ def test_single_mom_hook_failure_affects_vnode(self):
             id=single_mom.shortname, max_attempts=20, interval=2)
 
         vname = None
-        if self.is_cray is True:
-            vlist = []
-            vnl = self.server.filter(
-                VNODE, {'resources_available.vntype': 'cray_compute'})
-            vlist = vnl["resources_available.vntype=cray_compute"]
-            # Loop through each compute vnode in the list and check if state
-            # is offline
-            for v1 in vlist:
-                if vname is None:
-                    vname = v1
-                # Check that the vnode is in offline state
-                self.server.expect(
-                    VNODE, {'state': 'offline'}, id=v1, max_attempts=3,
-                    interval=2)
-        elif cpuset_mom is True:
+        if cpuset_mom is True:

```

Comment:
I think the cpuset chunk of code can move up to your function.  You need to create a list from what status() returns.  This is easily done via list comprehension.  You do it by: vlist = [x['id'] for x in vnl if x['id'] != single_mom.shortname]
If you want to know what that black magic is, I'll walk you through it on a whiteboard.